airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -434,9 +439,20 @@ def read(self, fields: Iterable = None) -> Iterable[dict]:
                 logger.error(f"{error_msg}\n{traceback.format_exc()}")
                 raise AirbyteTracedException(message=error_msg, internal_message=error_msg, failure_type=FailureType.config_error) from err
 
+    def _unzip(self, fp):
+        tmp_dir = tempfile.TemporaryDirectory()
+        with zipfile.ZipFile(str(fp.name), 'r') as zip_ref:
+            zip_ref.extractall(tmp_dir.name)
+            
+        logger.info("Temp dir content: " + str(os.listdir(tmp_dir.name)))
+        final_file:str =  os.path.join(tmp_dir.name, os.listdir(tmp_dir.name)[0])
+        logger.info("Pick up first file: " + final_file)
+        fp_tmp = open(final_file, 'r')
+        return fp_tmp
+
     def _cache_stream(self, fp):
         """cache stream to file"""
-        fp_tmp = tempfile.TemporaryFile(mode="w+b")
+        fp_tmp = tempfile.NamedTemporaryFile(mode="w+b")
```

Comment:
Any reason for this change?

Reply:
zipfile.ZipFile need a file path as string. TemporaryFile does not provide a path.